File: src/00125_ValidPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        s= "".join(ch for ch in s if ch.isalnum())
        strLen = len(s)

        if strLen <= 1:
            return True

        i = 0
        j = strLen - 1

        while (i < j):
            if s[i] != s[j]:
                return False
            
            i += 1
            j -= 1

        return True


# import math

# class Solution:
#     def isPalindrome(self, s: str) -> bool:
#         s = s.lower()
#         s= "".join(ch for ch in s if ch.isalnum())
#         print(s)
#         strLen = len(s)

#         if strLen <= 1:
#             return True

#         middle = math.ceil(strLen/2) - 1
#         if strLen%2 == 0:
#             p1 = middle 
#             p2 = middle + 1
#         else:
#             p1 = middle - 1
#             p2 = middle + 1

#         for i in range(p2, strLen):
#             if s[p1] != s[p2]:
#                 return False
#             else:
#                 p1 -= 1
#                 p2 += 1

#         return True

File: src/C#/00125_ValidPalindrome.cs
public class Solution {
    public bool IsPalindrome(string s) 
    {
        // Double pointer
        // Pointer left run from (middle=> 0)
        // Pointer right run from (middle => end)
        // Check each character 
        // Return true if all match, false otherwise
        var input = GetAlphanumericString(s).ToLower();
        int leftPt = 0;
        int rightPt = 0;

        if (input.Length % 2 == 0)
        {
            leftPt = input.Length/2 - 1;
            rightPt = leftPt + 1;
        }
        else
        {
            leftPt = input.Length/2 - 1;
            rightPt = leftPt + 2;
        }

        for (; rightPt < input.Length; rightPt++)
        {
            if (input[rightPt] != input[leftPt])
            {
                return false;
            }

            leftPt--;
        }

        return true;
    }

    private string GetAlphanumericString(string input)
    {
        return new String(input.Where(x => char.IsLetterOrDigit(x)).ToArray());
    }
}